workflows/scripts/convert.py

```diff
@@ -0,0 +1,61 @@
+"""
+script: covert.py
+
+converts sqlite (or other formats) into parquet files
+"""
+import pathlib
+from typing import Union, List
+
+import cytotable
+
+
+def sqlite_to_parquet(
+    input_file: Union[str, List[str]],
+    out_path: str,
+    target_ext: str,
+    convert_configs: str,
+):
+    # checking if user has parquet file
+    if target_ext == ".parquet" and pathlib.Path(input_file).suffix != ".sqlite":
+        raise ValueError(
+            "Converting to parquet files requires sqlite file."
+            f"File provided: {target_ext}"
+        )
+
+    # convert sqlite file into parquet
+    cytotable_config = convert_configs["params"]
+    cytotable.convert(
+        source_path=input_file,
+        dest_path=out_path,
+        dest_datatype=cytotable_config["dest_datatype"],
+        source_datatype=cytotable_config["source_datatype"],
+        concat=cytotable_config["concat"],
+        join=cytotable_config["join"],
+        infer_common_schema=cytotable_config["infer_common_schema"],
+        drop_null=cytotable_config["drop_null"],
+        preset=cytotable_config["preset"],
+        log_level=cytotable_config["log_level"],
+    )
+
+
+def main():
+    """Execution of the main script"""
+
+    # grabbing snakemake inputs
+    plate_data = str(snakemake.input)
+    output_path = str(snakemake.output)
+    general_configs = snakemake.params["data_configs"]
+    cytotable_configs = snakemake.params["cytotable_config"]
```

Comment:
Sounds good! here's also [documentation](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#python) on how `snakemake` gets imported into scripts.

Reply:
Given Snakemake's specificity and how it differs from other workflow implementations, it may be worth linking to that documentation within the code via a comment or within this project's documentation somehow. I respect your best judgement on this.